web/ASC.Web.Editor/src/helpers/withDialogs.js

```diff
@@ -0,0 +1,342 @@
+import React, { useState } from "react";
+import DynamicComponent from "../components/DynamicComponent";
+import { getPresignedUri } from "@appserver/common/api/files";
+import {
+  FILES_REMOTE_ENTRY_URL,
+  FILES_SCOPE,
+  STUDIO_SCOPE,
+  STUDIO_REMOTE_ENTRY_URL,
+} from "./constants";
+import Text from "@appserver/components/text";
+import TextInput from "@appserver/components/text-input";
+import Checkbox from "@appserver/components/checkbox";
+import { StyledSelectFolder } from "../StyledEditor";
+import { useTranslation } from "react-i18next";
+
+const insertImageAction = "imageFileType";
+const mailMergeAction = "mailMergeFileType";
+const compareFilesAction = "documentsFileType";
+
+const withDialogs = (WrappedComponent) => {
+  return (props) => {
+    const [isVisible, setIsVisible] = useState(false);
+    const [filesType, setFilesType] = useState("");
+    const [isFileDialogVisible, setIsFileDialogVisible] = useState(false);
+    const [typeInsertImageAction, setTypeInsertImageAction] = useState();
+    const [isFolderDialogVisible, setIsFolderDialogVisible] = useState(false);
+    const [titleSelectorFolder, setTitleSelectorFolder] = useState("");
+    const [urlSelectorFolder, setUrlSelectorFolder] = useState("");
+    const [extension, setExtension] = useState();
+    const [openNewTab, setNewOpenTab] = useState(false);
+    const [
+      preparationPortalDialogVisible,
+      setPreparationPortalDialogVisible,
+    ] = useState(false);
+
+    const { t } = useTranslation(["Editor", "Common"]);
+
+    const { fileInfo, fileId, mfReady } = props;
+    React.useEffect(async () => {
```

Comment:
Fix this "React Hook Warnings for async function in useEffect: useEffect function must return a cleanup function or nothing"